src/plugins/chart_expressions/expression_metric/public/components/metric_vis.tsx

```diff
@@ -0,0 +1,236 @@
+  const primaryMetricColumn = getColumnByAccessor(config.dimensions.metric, data.columns)!;
+  const formatPrimaryMetric = getFormatter(config.dimensions.metric, data.columns);
+
+  let secondaryMetricColumn;
+  let formatSecondaryMetric: ReturnType<typeof getFormatter>;
+  if (config.dimensions.secondaryMetric) {
+    secondaryMetricColumn = getColumnByAccessor(config.dimensions.secondaryMetric, data.columns);
+    formatSecondaryMetric = getFormatter(config.dimensions.secondaryMetric, data.columns);
+  }
+
+  const breakdownByColumn = config.dimensions.breakdownBy
+    ? getColumnByAccessor(config.dimensions.breakdownBy, data.columns)
+    : undefined;
+
+  const metricConfigs: MetricSpec['data'][number] = [];
+
+  const useProgressBar =
+    typeof config.metric.progressMin === 'number' && typeof config.metric.progressMax === 'number';
+
+  const commonProps = {
+    valueFormatter: formatPrimaryMetric,
+    ...(useProgressBar
+      ? {
+          domain: { min: config.metric.progressMin, max: config.metric.progressMax },
+          progressBarDirection: config.metric.progressDirection,
+        }
+      : {}),
+  };
+
+  if (!breakdownByColumn) {
+    const value = data.rows[0][primaryMetricColumn.id];
+
+    metricConfigs.push({
+      ...commonProps,
+      value,
+      title: primaryMetricColumn.name,
+      subtitle: secondaryMetricColumn?.name ?? config.metric.subtitle,
+      extra: (
+        <span>
+          {secondaryMetricColumn
+            ? formatSecondaryMetric!(data.rows[0][secondaryMetricColumn.id])
+            : config.metric.extraText}
+        </span>
+      ),
+      color: getColor(value, config.metric.palette),
+    });
+  }
+
+  if (breakdownByColumn) {
+    for (const row of data.rows) {
+      const value = row[primaryMetricColumn.id];
+
+      metricConfigs.push({
+        ...commonProps,
+        value,
+        title: row[breakdownByColumn.id],
+        subtitle: primaryMetricColumn.name,
+        extra: (
+          <span>
+            {secondaryMetricColumn
+              ? formatSecondaryMetric!(row[secondaryMetricColumn.id])
+              : config.metric.extraText}
+          </span>
+        ),
+        color: getColor(value, config.metric.palette),
+      });
+    }
+  }
+
+  if (config.metric.minTiles) {
+    while (metricConfigs.length < config.metric.minTiles) metricConfigs.push(undefined);
+  }
+
+  const grid: MetricSpec['data'] = [];
+  const {
+    metric: { maxCols },
+  } = config;
+  for (let i = 0; i < metricConfigs.length; i += maxCols) {
+    grid.push(metricConfigs.slice(i, i + maxCols));
+  }
+
+  const chartTheme = getThemeService().useChartsTheme();
+  const onRenderChange = useCallback<RenderChangeListener>(
+    (isRendered) => {
+      if (isRendered) {
+        renderComplete();
+      }
+    },
+    [renderComplete]
+  );
+
+  return (
+    <Chart>
+      <Settings
+        theme={[{ background: { color: 'transparent' } }, chartTheme]}
```

Comment:
This is somewhat safe for those charts because each tile has its own background color and we can compute correctly the high-contrast text color, but I think we should find a solution for Canvas soon
cc @flash1293